src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1052,6 +1053,12 @@ internal void DeleteModelInternal(List<ModelBase> modelsToDelete)
                 var node = model as NodeModel;
                 if (node != null)
                     node.Dispose();
+                var note = model as NoteModel;
+                if(note != null)
+                    note.Dispose();
+                var annotation = model as AnnotationModel;
+                if (annotation != null)
+                    annotation.Dispose();                
```

Comment:
Just out of curiosity, can we just have `ModelBase` implement `Dispose`?  This has a funny code smell.